blueocean-dashboard/src/main/js/PipelineRoutes.jsx

```diff
@@ -14,34 +12,72 @@ import {
     RunDetailsArtifacts,
     RunDetailsTests,
 } from './components';
+import { relativeUrl } from './util/UrlUtils';
+import Extensions from '@jenkins-cd/js-extensions';
+
+const DynamicRoutes = {
+    path: ':pipeline/(.*)',
+
+    getChildRoutes(partialNextState, callback) {
+        Extensions.store.getExtensions('pipeline.routes', routes => {
+            callback(null, routes); // routes is array
+        });
+    },
+};
+
+function goTo(path) {
+    let next = path;
+    while (next.charAt(0) === '/') {
+        next = next.substring(1);
+    }
+    return {
+        onEnter: ({ location, params }, replace) => {
+            replace(relativeUrl(location, next));
+        },
+    };
+}
+
+// multibranch-pr-github/detail/test-pass-fail-stage/1/pipeline
+// :pipeline/detail/:branch/:runId/pipeline
+const runDetailRoutes = { path: ':runId', component: RunDetails,
```

Comment:
Does this subsume some of what was sketched in https://github.com/jenkinsci/blueocean-plugin/pull/362 ?